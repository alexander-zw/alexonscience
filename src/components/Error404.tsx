import "../styles/index.css";

import React from "react";
import { Link } from "react-router-dom";

import MetaTags from "./subcomponents/MetaTags";

/**
 * This is the 404 Error page.
 */
function Error404() {
    return (
        <div className="text-div">
            <MetaTags path="/error" />
            <p>
                Oops, this page does not exist! <Link to="/">Back to Home</Link>
            </p>
        </div>
    );
}

export default Error404;
