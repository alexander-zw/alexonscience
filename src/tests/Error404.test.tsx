import "@testing-library/jest-dom";

import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import Error404 from "../components/Error404";

/**
 * Test for the Error404.tsx component, which is essentially the 404 page.
 */
describe("Error404", () => {
    test("renders correctly", () => {
        // Error has a Link, so we surround it with a router.
        const tree = renderer
            .create(
                <BrowserRouter>
                    <Error404 />
                </BrowserRouter>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
