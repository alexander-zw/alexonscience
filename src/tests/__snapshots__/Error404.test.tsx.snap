// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 renders correctly 1`] = `
<div
  className="text-div"
>
  <p>
    Oops, this page does not exist! 
    <a
      href="/"
      onClick={[Function]}
    >
      Back to Home
    </a>
  </p>
</div>
`;
